app/src/main/java/com/hedvig/app/feature/trustly/TrustlyConnectFragment.kt

```diff
@@ -107,14 +111,19 @@ class TrustlyConnectFragment : Fragment(R.layout.trustly_connect_fragment) {
                 webViewClient = object : WebViewClient() {
                     override fun onPageFinished(view: WebView?, url: String?) {
                         super.onPageFinished(view, url)
-                        if (!hasLoadedWebView) {
-                            TransitionManager.beginDelayedTransition(
-                                binding.root,
-                                MaterialFadeThrough()
-                            )
-                            loadingContainer.isVisible = false
-                            trustly.isVisible = true
-                            hasLoadedWebView = true
+                        lifecycleScope.launch {
+                            // Need to check lifecycle here since we are starting other activities in onPageStarted
+                            viewLifecycleOwner.withStateAtLeast(Lifecycle.State.CREATED) {
+                                if (!hasLoadedWebView) {
```

Comment:
I added some lifecycle checks here. Do you think this could work?

Reply:
Probably. Maybe we can simplify it to `if (viewLifecycleOwner.lifecycle.currentState.isAtLeast(Lifecycle.State.Created) { ... }`? I worry that creating a coroutine scope here might complicate things even more.

Reply:
Sounds good